pkg/util/informermanager/federatedinformermanager.go

```diff
@@ -90,28 +99,33 @@ func NewFederatedInformerManager(
 	clusterInformer fedcorev1a1informers.FederatedClusterInformer,
 ) FederatedInformerManager {
 	manager := &federatedInformerManager{
-		lock:                   sync.RWMutex{},
-		started:                false,
-		shutdown:               false,
-		clientHelper:           clientHelper,
-		ftcInformer:            ftcInformer,
-		clusterInformer:        clusterInformer,
-		eventHandlerGenerators: []*EventHandlerGenerator{},
-		clusterEventHandlers:   []*ClusterEventHandler{},
-		kubeClients:            map[string]kubernetes.Interface{},
-		dynamicClients:         map[string]dynamic.Interface{},
-		restConfigs:            map[string]*rest.Config{},
-		connectionMap:          map[string][]byte{},
-		clusterCancelFuncs:     map[string]context.CancelFunc{},
-		informerManagers:       map[string]InformerManager{},
-		informerFactories:      map[string]informers.SharedInformerFactory{},
+		lock:                     sync.RWMutex{},
+		started:                  false,
+		shutdown:                 false,
+		clientHelper:             clientHelper,
+		ftcInformer:              ftcInformer,
+		clusterInformer:          clusterInformer,
+		eventHandlerGenerators:   []*EventHandlerGenerator{},
+		clusterEventHandlers:     []*ClusterEventHandler{},
+		kubeClients:              map[string]kubernetes.Interface{},
+		dynamicClients:           map[string]dynamic.Interface{},
+		restConfigs:              map[string]*rest.Config{},
+		connectionMap:            map[string][]byte{},
+		clusterCancelFuncs:       map[string]context.CancelFunc{},
+		informerManagers:         map[string]InformerManager{},
+		informerFactories:        map[string]informers.SharedInformerFactory{},
+		dynamicInformerFactories: map[string]dynamicinformer.DynamicSharedInformerFactory{},
 		queue: workqueue.NewRateLimitingQueue(
 			workqueue.NewItemExponentialFailureRateLimiter(100*time.Millisecond, 10*time.Second),
 		),
-		podListerSemaphore:    semaphore.NewWeighted(3), // TODO: make this configurable
-		initialClusters:       sets.New[string](),
-		podEventHandlers:      []*ResourceEventHandlerWithClusterFuncs{},
-		podEventRegistrations: map[string]map[*ResourceEventHandlerWithClusterFuncs]cache.ResourceEventHandlerRegistration{},
+		podListerSemaphore:              semaphore.NewWeighted(3), // TODO: make this configurable
+		initialClusters:                 sets.New[string](),
+		podEventHandlers:                []*ResourceEventHandlerWithClusterFuncs{},
+		podEventRegistrations:           map[string]map[*ResourceEventHandlerWithClusterFuncs]cache.ResourceEventHandlerRegistration{},
+		serviceExportEventHandlers:      []*FilteringResourceEventHandlerWithClusterFuncs{},
```

Comment:
InformerManager is shared across all components and ideally should not contain hardcoded resource-specific logic like this (the pod case above is a special case due to the volume of pods in large clusters). Any chance to abstract this further and shield InformerManager from controller-specific logic?

Reply:
For dynamicInformerFactory, I did consider encapsulating a generic one. The reason for using this form is firstly because of the reference to the above-mentioned podEventHandlers, and secondly because this structure is already a bit complicated, map[string]map[*ResourceEventHandlerWithClusterFuncs]cache.ResourceEventHandlerRegistration{}, and adding another layer up will make it even more complicated. If the pod case is a special case, I think a generic logic is ok.

Reply:
I agreed to use a common interfaces for dynamic resources. For native resources(such as endpointSlice or service), I prefer to add a separate interface. Because even if we use a common interface, we have to implement it by a switch-case logic. 
```
case endpointSliceGVK:
    factory.Discovery.V1beta1().EndpointSlice()....
case serviceGVK:
    factory.Core().V1().Services()...
```